src/renderer/port-forward/port-forward-store/request-active-port-forward.injectable.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import { getInjectable } from "@ogre-tools/injectable";
+import { apiBaseInjectionToken } from "../../../common/k8s-api/api-base";
+import loggerInjectable from "../../../common/logger.injectable";
+import { urlBuilderFor } from "../../../common/utils/buildUrl";
+import type { ForwardedPort } from "../port-forward-item";
+
+export type RequestActivePortForward = (portForward: ForwardedPort) => Promise<ForwardedPort | undefined>;
+
+const requestActiveEndpoint = urlBuilderFor("/pods/port-forward/:namespace/:kind/:name");
+
+const requestActivePortForwardInjectable = getInjectable({
+  id: "request-active-port-forward",
+  instantiate: (di): RequestActivePortForward => {
+    const apiBase = di.inject(apiBaseInjectionToken);
+    const logger = di.inject(loggerInjectable);
+
+    return async (portForward) => {
+      const { port, forwardPort } = portForward;
+      let response: { port: number };
+
+      try {
+        response = await apiBase.get(requestActiveEndpoint.compile(portForward), { query: { port, forwardPort }});
+      } catch (error) {
+        logger.warn(`[PORT-FORWARD-STORE] Error getting active port-forward: ${error}`, portForward);
+
+        return undefined;
+      }
+
+      portForward.status = response?.port ? "Active" : "Disabled";
```

Comment:
Generally, mutation of an input parameter is a pretty bad idea, as it's easy to cause bugs that way. There's even a lint for this.

Reply:
I completely agree but I mostly just moved this code around